encryption/src/main/kotlin/io/provenance/scope/encryption/model/Key.kt

```diff
@@ -1,9 +1,18 @@
 package io.provenance.scope.encryption.model
 
+import io.provenance.scope.encryption.crypto.Pen
+import io.provenance.scope.encryption.crypto.SignerImpl
+import java.security.KeyPair
 import java.security.PrivateKey
 import java.security.PublicKey
 import java.util.UUID
 
 sealed class KeyRef(val publicKey: PublicKey)
 class SmartKeyRef(publicKey: PublicKey, val uuid: UUID) : KeyRef(publicKey)
 class DirectKeyRef(publicKey: PublicKey, val privateKey: PrivateKey) : KeyRef(publicKey)
+
+// TODO implement smart key leg
+fun KeyRef.signer(): SignerImpl = when (this) {
```

Comment:
I think we will still need a SignerFactory instead of an extension function, since the SmartKey implementation will need the SmartKey api provided to it... but I know @wbaker-figure has a pr with SmartKey stuff up and I haven't yet gone through that

Reply:
nevermind, we discussed this on the other PR, probably going this way and putting the api reference in the SmartKeyRef object is the best way to go instead of the factory that injects the api